### DISCUSSION TITLE
[Testing] Zustand testing and/or mocking process failed in React Native

### DISCUSSION THREAD

**Discussion by jimmynguyen1308** [pmndrs/zustand#1897] [Bug report] [answered]
Hi!

I am currently working on a React Native project that using Zustand (v4.3.8) for state management and using `vitest` for testing. 

While I was trying to mock Zustand in accordance to [this documentation](https://docs.pmnd.rs/zustand/guides/testing), I encountered an error `TypeError: create(...) is not a function`.

At first I thought it might be some configuration conflicts with React Native and/or Vitest; however, the exact bug can be found in Zustand example [on CodeSandbox](https://codesandbox.io/p/sandbox/zustand-vitest-demo-ph5gnj?file=%2Fsrc%2Fcomponents%2Fcounter%2Fcounter.test.tsx%3A32%2C1)

Screenshot:

<img width="1504" alt="Screenshot 2023-06-27 at 12 13 50 pm" src="https://github.com/pmndrs/zustand/assets/118421449/566442cd-d16f-484b-b33c-147ba1083566">

It might be related to issue #1505, but I am new to Zustand so I can't be sure. For reference, my code for one store looks like this...
```typescript
import { create, StateCreator } from "zustand"
import { persist, createJSONStorage } from "zustand/middleware"
import AsyncStorage from "@react-native-async-storage/async-storage"

export interface IMessageStore {
  message: string | null
  setMessage: (message: string | null) => void
}

export const messageState: StateCreator<IMessageStore> = (set) => ({
  message: null,
  setMessage: (message: string | null) => set({ message }),
})

export const useMessageStore = create<IMessageStore>()(
  persist(
    (set, get, api) => ({
      ... messageState(set, get, api),
    }),
    {
      name: "message-storage",
      storage: createJSONStorage(() => AsyncStorage),
    }
  )
)
```

**Comment by dbritto-dev** [comment]
@jimmynguyen1308 thanks for pointing out that I'll take a look of that

**Reply by dbritto-dev** [reply]
@jimmynguyen1308 I updated the example

**Reply by jimmynguyen1308** [reply]
Thanks. I'll have a look.

**Reply by dbritto-dev** [reply]
@jimmynguyen1308 keep me in the loop

**Comment by jimmynguyen1308** [comment]
Found the issue on my end. Somehow my project isn't compatible with `@testing-library/jest-dom`, so I have to put the mock directly to the `__mock__` path location:
```typescript
// setup-vitest.ts
import { vi } from "vitest"
vi.mock("./src/__mock__/zustand")
```

Not too sure if it's the same case with the CodeSandbox, but thanks @dbritto-dev for your quick response. Cheers!

**Reply by dbritto-dev** [reply, marked as answer]
@jimmynguyen1308 you shouldn't do that please look into latest codesandbox demo. I updated the docs too. https://github.com/pmndrs/zustand/pull/1898/files
